lib/spack/spack/test/flag_handlers.py

```diff
@@ -37,6 +39,14 @@ def temp_env():
     os.environ = old_env
 
 
+@pytest.fixture()
+def mock_repo():
+    repo_path = spack.repo.RepoPath(spack.paths.mock_packages_path)
+    mock_repo = copy.deepcopy(repo_path)
```

Comment:
I have two questions on this line:
1. Why do we need to deep copy the object?
2. Can't we avoid to have `mock_repo` and use the already available `mock_packages` fixture instead?

Reply:
TBH, I just copy-and-pasted this from `conftest.py`. I had not used pytest before and did not know that it is this easy to use fixtures in other files. Thanks for the hints!

I have pushed an updated commit that simply uses the `mock_packages` fixture without changing all signatures.